### PR TITLE
Update jQuery 2.2 and 3.3.1 in samples to 3.4.1

### LINKED ISSUE #20001: Consider updating jQuery dependency to 3.4.0 or newer
My team runs Azure DevOps Component Governance (per internal security requirements) and we received a notice that the following project template is impacted by [CVE-2019-11358](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-11358) affecting jQuery before 3.4.0.

```
cli/sdk/2.2.100-preview3-009430/Templates/microsoft.dotnet.web.projecttemplates.2.2.2.2.0-preview3-35497.nupkg
```

I looked inside this template and found jQuery 3.3.1.

I also looked at the latest version of the template and the source code in this repository and see that 3.3.1 is still used: e.g.
https://github.com/dotnet/aspnetcore/blob/7def10277abbb3086c4ee00e369faf6aa27d676a/src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/wwwroot/lib/jquery/dist/jquery.js#L2

We don't specifically use this template but Component Governance picked up on it since we happened to install the CLI at build time inside the directory that it scans.

It's probably worth considering updating the template to jQuery 3.4.0 so consumers of the template don't have to be worried about this CVE. 

I see there is another issue of removing jQuery (https://github.com/dotnet/aspnetcore/issues/8573) but I imagine updating the minor version of jQuery is a cheaper fix for this particular issue.

### PR DESCRIPTION
**Pull request by wenz**
Updates outdated insecure jQuery versions to latest release 3.4.1

Addresses #20001

### CODE REVIEW

**Comment by wenz** [comment]
In those cases where I was adding the jquery.min.js file, it was just missing from the sample, but still referenced using the tag helper fallback.

File: src/Azure/AzureAD/samples/AzureADB2CSample/Pages/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -51,11 +51,11 @@
         <script src="~/js/site.js" asp-append-version="true"></script>
     </environment>
     <environment exclude="Development">
-        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-2.2.0.min.js"
+        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-3.4.1.min.js"
                 asp-fallback-src="~/lib/jquery/dist/jquery.min.js"
                 asp-fallback-test="window.jQuery"
                 crossorigin="anonymous"
-                integrity="sha384-K+ctZQ+LL8q6tP7I94W+qzQsfRV2a+AfHIi9k8z8l9ggpc8X+Ytst4yBo/hH+8Fk">
+                integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=">
```
Comment (anurse) [inline src/Azure/AzureAD/samples/AzureADB2CSample/Pages/Shared/_Layout.cshtml]: FYI: Verified SRI hash independently.

File: src/Azure/AzureAD/samples/AzureADSample/Pages/Shared/_Layout.cshtml [hunk 2]
```diff
@@ -51,11 +51,11 @@
         <script src="~/js/site.js" asp-append-version="true"></script>
     </environment>
     <environment exclude="Development">
-        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-2.2.0.min.js"
+        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-3.4.1.min.js"
                 asp-fallback-src="~/lib/jquery/dist/jquery.min.js"
                 asp-fallback-test="window.jQuery"
                 crossorigin="anonymous"
-                integrity="sha384-K+ctZQ+LL8q6tP7I94W+qzQsfRV2a+AfHIi9k8z8l9ggpc8X+Ytst4yBo/hH+8Fk">
+                integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=">
```
Comment (anurse) [inline src/Azure/AzureAD/samples/AzureADSample/Pages/Shared/_Layout.cshtml]: FYI: Verified SRI hash independently.

File: src/Identity/UI/src/Areas/Identity/Pages/V3/_Layout.cshtml [hunk 3]
```diff
@@ -63,11 +63,11 @@
         <script src="~/Identity/js/site.js" asp-append-version="true"></script>
     </environment>
     <environment exclude="Development">
-        <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"
+        <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js"
                 asp-fallback-src="~/Identity/lib/jquery/dist/jquery.min.js"
                 asp-fallback-test="window.jQuery"
                 crossorigin="anonymous"
-                integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=">
+                integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=">
```
Comment (anurse) [inline src/Identity/UI/src/Areas/Identity/Pages/V3/_Layout.cshtml]: FYI: Verified SRI hash independently.

File: src/Identity/UI/src/Areas/Identity/Pages/V4/_Layout.cshtml [hunk 4]
```diff
@@ -74,11 +74,11 @@
         <script src="~/Identity/js/site.js" asp-append-version="true"></script>
     </environment>
     <environment exclude="Development">
-        <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"
+        <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js"
                 asp-fallback-src="~/Identity/lib/jquery/dist/jquery.min.js"
                 asp-fallback-test="window.jQuery"
                 crossorigin="anonymous"
-                integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=">
+                integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=">
```
Comment (anurse) [inline src/Identity/UI/src/Areas/Identity/Pages/V4/_Layout.cshtml]: FYI: Verified SRI hash independently.

File: src/Identity/testassets/Identity.DefaultUI.WebSite/Pages/_Layout.cshtml [hunk 5]
```diff
@@ -56,11 +56,11 @@
         <script src="~/lib/bootstrap/dist/js/bootstrap.bundle.js"></script>
     </environment>
     <environment exclude="Development">
-        <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"
+        <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js"
                 asp-fallback-src="~/lib/jquery/dist/jquery.min.js"
                 asp-fallback-test="window.jQuery"
                 crossorigin="anonymous"
-                integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=">
+                integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=">
```
Comment (anurse) [inline src/Identity/testassets/Identity.DefaultUI.WebSite/Pages/_Layout.cshtml]: FYI: Verified SRI hash independently.

File: src/Mvc/samples/MvcSandbox/Views/Shared/_Layout.cshtml [hunk 6]
```diff
@@ -33,7 +33,7 @@
         </main>
     </div>
 
-    <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
+    <script src="https://code.jquery.com/jquery-3.4.1.slim.min.js" integrity="sha256-pasqAKBDmFT4eHoN2ndd6lN370kFiGUFyTiUHWhU7k8=" crossorigin="anonymous"></script>
```
Comment (anurse) [inline src/Mvc/samples/MvcSandbox/Views/Shared/_Layout.cshtml]: FYI: Verified SRI hash independently.

File: src/Security/samples/ClaimsTransformation/Views/Shared/_Layout.cshtml [hunk 7]
```diff
@@ -54,11 +54,11 @@
         <script src="~/js/site.js" asp-append-version="true"></script>
     </environment>
     <environment exclude="Development">
-        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-2.2.0.min.js"
+        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-3.4.1.min.js"
                 asp-fallback-src="~/lib/jquery/dist/jquery.min.js"
                 asp-fallback-test="window.jQuery"
                 crossorigin="anonymous"
-                integrity="sha384-K+ctZQ+LL8q6tP7I94W+qzQsfRV2a+AfHIi9k8z8l9ggpc8X+Ytst4yBo/hH+8Fk">
+                integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=">
```
Comment (anurse) [inline src/Security/samples/ClaimsTransformation/Views/Shared/_Layout.cshtml]: FYI: Verified SRI hash independently.

File: src/Security/samples/Cookies/Views/Shared/_Layout.cshtml [hunk 8]
```diff
@@ -54,11 +54,11 @@
         <script src="~/js/site.js" asp-append-version="true"></script>
     </environment>
     <environment exclude="Development">
-        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-2.2.0.min.js"
+        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-3.4.1.min.js"
                 asp-fallback-src="~/lib/jquery/dist/jquery.min.js"
                 asp-fallback-test="window.jQuery"
                 crossorigin="anonymous"
-                integrity="sha384-K+ctZQ+LL8q6tP7I94W+qzQsfRV2a+AfHIi9k8z8l9ggpc8X+Ytst4yBo/hH+8Fk">
+                integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=">
```
Comment (anurse) [inline src/Security/samples/Cookies/Views/Shared/_Layout.cshtml]: FYI: Verified SRI hash independently.

File: src/Security/samples/DynamicSchemes/Pages/_Layout.cshtml [hunk 9]
```diff
@@ -51,11 +51,11 @@
         <script src="~/js/site.js" asp-append-version="true"></script>
     </environment>
     <environment exclude="Development">
-        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-2.2.0.min.js"
+        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-3.4.1.min.js"
                 asp-fallback-src="~/lib/jquery/dist/jquery.min.js"
                 asp-fallback-test="window.jQuery"
                 crossorigin="anonymous"
-                integrity="sha384-K+ctZQ+LL8q6tP7I94W+qzQsfRV2a+AfHIi9k8z8l9ggpc8X+Ytst4yBo/hH+8Fk">
+                integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=">
```
Comment (anurse) [inline src/Security/samples/DynamicSchemes/Pages/_Layout.cshtml]: FYI: Verified SRI hash independently.

File: src/Security/samples/Identity.ExternalClaims/Pages/_Layout.cshtml [hunk 10]
```diff
@@ -52,11 +52,11 @@
         <script src="~/js/site.js" asp-append-version="true"></script>
     </environment>
     <environment exclude="Development">
-        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-2.2.0.min.js"
+        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-3.4.1.min.js"
                 asp-fallback-src="~/lib/jquery/dist/jquery.min.js"
                 asp-fallback-test="window.jQuery"
                 crossorigin="anonymous"
-                integrity="sha384-K+ctZQ+LL8q6tP7I94W+qzQsfRV2a+AfHIi9k8z8l9ggpc8X+Ytst4yBo/hH+8Fk">
+                integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=">
```
Comment (anurse) [inline src/Security/samples/Identity.ExternalClaims/Pages/_Layout.cshtml]: FYI: Verified SRI hash independently.

File: src/Security/samples/PathSchemeSelection/Views/Shared/_Layout.cshtml [hunk 11]
```diff
@@ -54,11 +54,11 @@
         <script src="~/js/site.js" asp-append-version="true"></script>
     </environment>
     <environment exclude="Development">
-        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-2.2.0.min.js"
+        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-3.4.1.min.js"
                 asp-fallback-src="~/lib/jquery/dist/jquery.min.js"
                 asp-fallback-test="window.jQuery"
                 crossorigin="anonymous"
-                integrity="sha384-K+ctZQ+LL8q6tP7I94W+qzQsfRV2a+AfHIi9k8z8l9ggpc8X+Ytst4yBo/hH+8Fk">
+                integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=">
```
Comment (anurse) [inline src/Security/samples/PathSchemeSelection/Views/Shared/_Layout.cshtml]: FYI: Verified SRI hash independently.

File: src/Security/samples/StaticFilesAuth/Views/Shared/_Layout.cshtml [hunk 12]
```diff
@@ -51,11 +51,11 @@
         <script src="~/js/site.js" asp-append-version="true"></script>
     </environment>
     <environment exclude="Development">
-        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-2.2.0.min.js"
+        <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-3.4.1.min.js"
                 asp-fallback-src="~/lib/jquery/dist/jquery.min.js"
                 asp-fallback-test="window.jQuery"
                 crossorigin="anonymous"
-                integrity="sha384-K+ctZQ+LL8q6tP7I94W+qzQsfRV2a+AfHIi9k8z8l9ggpc8X+Ytst4yBo/hH+8Fk">
+                integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=">
```
Comment (anurse) [inline src/Security/samples/StaticFilesAuth/Views/Shared/_Layout.cshtml]: FYI: Verified SRI hash independently.

**Review by analogrelay — COMMENTED** [review]
Hrm, I'm trying to verify the bundled `jquery.js`, `jquery.min.js` and `jquery.min.map` files in this commit and I'm seeing the file hashes don't seem to match the ones in the bower or npm packages:

In the commit:

```
Name           Hash
----           ----
jquery.js      54534731E9176376051FC2F721D23F8528863756DB0B50B80572C91FBE4F4688
jquery.min.js  412B8FF9C5AB32B9019FCD84BCD4A54C0E265A14528474F4EE45B27A20ABEAEB
jquery.min.map 8DA74AEC0FCDD7678A2663B3CC9BAFBAF009E6D6929B28BB3DD95BCED18206F6
```

From https://github.com/jquery/jquery-dist/tree/master/dist and npm (both sources have the same hashes):

```
Name                Hash                                                             Algorithm
----                ----                                                             ---------
core.js             DFA8443DC1BC98AA2891FBD456F12A464A5D615D6E9CFEB0BF122A6E075E5619 SHA256
jquery.js           5A93A88493AA32AAB228BF4571C01207D3B42B0002409A454D404B4D8395BD55 SHA256
jquery.min.js       0925E8AD7BD971391A8B1E98BE8E87A6971919EB5B60C196485941C3C1DF089A SHA256
jquery.min.map      DB5EA0749F63CA5F4E2EB34D8F6F197D04BDDA1F633C72A0C77CB4BACFD484B1 SHA256
jquery.slim.js      0539537503BDFDF6AC701D5DADE92B0D591A29DF4F93007298C9473A21BEA8B2 SHA256
jquery.slim.min.js  A5AB2A00A0439854F8787A0DDA775DEA5377EF4905886505C938941D6854EE4F SHA256
jquery.slim.min.map 9A41730CFAE1CBAEC9EBEBB4649E7F616236B512AC37C1107A9CEBE4096EDE3C SHA256
```

@wenz can you confirm how you updated the bundled jquery files (i.e. the checked-in `jquery.js`, `jquery.min.js` and `jquery.min.map` files)? I'm sure it's just a conflict in how I'm verifying these (gotta be careful though 😄).

The SRI hashes all look accurate and represent the actual hash of the content on the aspnetcdn, so those are good.

**Comment by wenz** [comment]
I downloaded the latest versions from https://jquery.com/download/ - to be precise https://code.jquery.com/jquery-3.4.1.min.js, https://code.jquery.com/jquery-3.4.1.js, https://code.jquery.com/jquery-3.4.1.min.map, and in one instance, https://code.jquery.com/jquery-3.4.1.slim.min.js. I then put them in the repo. Maybe I messed up w/r/t line endings? I'll double-check later today!

**Comment by anurse** [comment]
> Maybe I messed up w/r/t line endings? I'll double-check later today!

That’s possible! I can verify from those sources too. I just wanted to sanity check that the files were correct. I’m not too worried which source they come from as long as it’s an official source and relatively easy to independently verify :).

**Comment by wenz** [comment]
So when calculating the SHA256 hashes on my local repository, I get the correct results, so the transport is the culprit - CR/LF for sure. I'll try to find out the easiest way to fix this. Thanks for noticing!

**Comment by anurse** [comment]
> So when calculating the SHA256 hashes on my local repository, I get the correct results, so the transport is the culprit - CR/LF for sure.

Ok, I can also give it another try with normalizing line endings. I'm really not worried if the line endings are different (these are just samples), just want to make sure I can validate that the content is the same so I know we're getting what you say we're getting (I'm sure you're trustworthy, but it's important to check 😝)

**Comment by wenz** [comment]
By all means - we are fixing a security vulnerability here, so the files need to be 100% genuine. 

My Git-fu is limited, so if you can normalize on your end, I'd appreciate it ;) (Otherwise, I'll try to fix it tomorrow morning. Looks like my dev machine has core.autocrlf=true.) The samples will probably only work if the hashes match the files, so they need to be exact IMO.

**Comment by HaoK** [comment]
Should we take this in release/3.1 and merge forward to master?

**Comment by mkArtakMSFT** [comment]
> Should we take this in release/3.1 and merge forward to master?

Yes! But we can do the opposite too, can't we?

**Comment by anurse** [comment]
> Should we take this in release/3.1 and merge forward to master?

Yeah, I suppose. These are non-shipping samples, but should probably keep up-to-date even in `release/3.1`.

**Comment by HaoK** [comment]
This isn't only samples, see src/Identity/UI/src/Areas/Identity/Pages/V3/_Layout.cshtml

**Comment by HaoK** [comment]
ProjectTemplates and Identity product code have copies of jquery...  we probably also should update scaffolding at the same time: https://github.com/dotnet/Scaffolding/tree/master/src/VS.Web.CG.Mvc/Templates/Identity/Bootstrap4/wwwroot/lib

File: src/Identity/testassets/Identity.DefaultUI.WebSite/Pages/_Layout.cshtml (hunk 5 above)
Comment (HaoK) [reply src/Identity/testassets/Identity.DefaultUI.WebSite/Pages/_Layout.cshtml]: This one is product code

File: src/Identity/UI/src/Areas/Identity/Pages/V4/_Layout.cshtml (hunk 4 above)
Comment (HaoK) [reply src/Identity/UI/src/Areas/Identity/Pages/V4/_Layout.cshtml]: This one is product code

**Comment by anurse** [comment]
> This isn't only samples, see src/Identity/UI/src/Areas/Identity/Pages/V3/_Layout.cshtml

Ok, let's get this in `master` then and we can open a PR to propose backporting and take that through servicing.

**Comment by mkArtakMSFT** [comment]
@HaoK I think the scaffolding piece is something we should take care of. @wenz would you be willing to also update the ProjectTemplates and Identity areas too to reference the latest jQuery? It would be super helpful!

**Comment by wenz** [comment]
> @wenz would you be willing to also update the ProjectTemplates and Identity areas too to reference the latest jQuery? It would be super helpful!

Absolutely! Will create an issue and PR in the Scaffolding project once I've sorted my CRLF problems.

**Comment by wenz** [comment]
I've updated the PR; now the line endings are unaltered and the hashes match (at least in my tests). 

Thank you Git guru @TimBo93!

**Comment by wenz** [comment]
I just double-checked: 

> @HaoK I think the scaffolding piece is something we should take care of.

The Scaffolding project is already using jQuery 3.4.1, so no action required here IMO. 

> @wenz would you be willing to also update the ProjectTemplates and Identity areas too to reference the latest jQuery? It would be super helpful!

Already caught that with the first commit :-)

File: src/Azure/AzureAD/samples/AzureADB2CSample/wwwroot/lib/jquery/dist/jquery.min.map [hunk 13]
```diff
@@ -1 +1 @@
...
...4f,MAAMuO,SAAW,YAGvBsX,EAAYE,EAAQR,SACpBI,EAAY7mC,EAAOohB,IAAK9f,EAAM,OAC9B0lC,EAAahnC,EAAOohB,IAAK9f,EAAM,SACI,aAAbmuB,GAAwC,UAAbA,KACA,GAA9CoX,EAAYG,GAAanpC,QAAS,SAMpCipC,GADAH,EAAcM,EAAQxX,YACD5iB,IACrB+5B,EAAUD,EAAY3S,OAGtB8S,EAAShX,WAAY+W,IAAe,EACpCD,EAAU9W,WAAYkX,IAAgB,GAGlC1oC,EAAY4D,KAGhBA,EAAUA,EAAQ9D,KAAMkD,EAAMnC,EAAGa,EAAOiC,OAAQ,GAAI8kC,KAGjC,MAAf7kC,EAAQ2K,MACZ+b,EAAM/b,IAAQ3K,EAAQ2K,IAAMk6B,EAAUl6B,IAAQi6B,GAE1B,MAAhB5kC,EAAQ8xB,OACZpL,EAAMoL,KAAS9xB,EAAQ8xB,KAAO+S,EAAU/S,KAAS4S,GAG7C,UAAW1kC,EACfA,EAAQglC,MAAM9oC,KAAMkD,EAAMsnB,GAG1Bqe,EAAQ7lB,IAAKwH,KAKhB5oB,EAAOG,GAAG8B,OAAQ,CAGjBwkC,OAAQ,SAAUvkC,GAGjB,GAAKV,UAAUjB,OACd,YAAmBqC,IAAZV,EACN9E,KACAA,KAAK+D,KAAM,SAAUhC,GACpBa,EAAOymC,OAAOC,UAAWtpC,KAAM8E,EAAS/C,KAI3C,IAAIgoC,EAAMC,EACT9lC,EAAOlE,KAAM,GAEd,OAAMkE,EAQAA,EAAK6wB,iBAAiB5xB,QAK5B4mC,EAAO7lC,EAAKwyB,wBACZsT,EAAM9lC,EAAK2I,cAAc2C,YAClB,CACNC,IAAKs6B,EAAKt6B,IAAMu6B,EAAIC,YACpBrT,KAAMmT,EAAKnT,KAAOoT,EAAIE,cARf,CAAEz6B,IAAK,EAAGmnB,KAAM,QATxB,GAuBDvE,SAAU,WACT,GAAMryB,KAAM,GAAZ,CAIA,IAAImqC,EAAcd,EAAQvnC,EACzBoC,EAAOlE,KAAM,GACboqC,EAAe,CAAE36B,IAAK,EAAGmnB,KAAM,GAGhC,GAAwC,UAAnCh0B,EAAOohB,IAAK9f,EAAM,YAGtBmlC,EAASnlC,EAAKwyB,4BAER,CACN2S,EAASrpC,KAAKqpC,SAI
```
Comment (anurse) [inline src/Azure/AzureAD/samples/AzureADB2CSample/wwwroot/lib/jquery/dist/jquery.min.map:1]: These sourcemaps reference `jquery-3.4.1.min.js`, but the file we have is `jquery.min.js`, so they won't line up properly when used in the browser. This is also why the hashes still aren't matching up in my comparisons with the content in the `jquery@3.4.1` package on npm.

**Review by analogrelay — COMMENTED** [review]
Hashes look better but not 100% right. I validated `jquery.js` and `jquery.min.js` 🎉, but `jquery.min.map` isn't lining up with the content in the npm package. I added a comment as to why I think that is.

@wenz thanks for hanging in there while we get this right!

**Comment by anurse** [comment]
For reference, now the local hashes are:

```
Name           Hash
----           ----
jquery.js      54534731E9176376051FC2F721D23F8528863756DB0B50B80572C91FBE4F4688
jquery.min.js  412B8FF9C5AB32B9019FCD84BCD4A54C0E265A14528474F4EE45B27A20ABEAEB
jquery.min.map 8DA74AEC0FCDD7678A2663B3CC9BAFBAF009E6D6929B28BB3DD95BCED18206F6
```

And the ones in npm are (unchanged from above, but repeated):

```
Name                Hash                                                             Algorithm
----                ----                                                             ---------
jquery.js           54534731E9176376051FC2F721D23F8528863756DB0B50B80572C91FBE4F4688 SHA256
jquery.min.js       412B8FF9C5AB32B9019FCD84BCD4A54C0E265A14528474F4EE45B27A20ABEAEB SHA256
jquery.min.map      DB5EA0749F63CA5F4E2EB34D8F6F197D04BDDA1F633C72A0C77CB4BACFD484B1 SHA256
```

File: src/Azure/AzureAD/samples/AzureADB2CSample/wwwroot/lib/jquery/dist/jquery.min.map (hunk 13 above)
Comment (wenz) [reply src/Azure/AzureAD/samples/AzureADB2CSample/wwwroot/lib/jquery/dist/jquery.min.map:1]: Good catch! I'll alter the map files.

Comment (anurse) [reply src/Azure/AzureAD/samples/AzureADB2CSample/wwwroot/lib/jquery/dist/jquery.min.map:1]: We shouldn't have to alter them. They appear to be correct already in sources like the `npm` package. I'd suggest taking them from there instead.

Comment (wenz) [reply src/Azure/AzureAD/samples/AzureADB2CSample/wwwroot/lib/jquery/dist/jquery.min.map:1]: You're right, they are correct on npm (the official downloads use the versions file name, though). I'm on it!

**Comment by mkArtakMSFT** [comment]
@anurse let's try to get this into `master` and then we will need to bring this into both 3.1 and 2.1 as well, but we will need to handle that (@HaoK).

**Comment by anurse** [comment]
Yep, that's the plan.

File: src/Azure/AzureAD/samples/AzureADB2CSample/wwwroot/lib/jquery/dist/jquery.min.map (hunk 13 above)
Comment (wenz) [reply src/Azure/AzureAD/samples/AzureADB2CSample/wwwroot/lib/jquery/dist/jquery.min.map]: @anurse Just added the .map files.

**Comment by anurse** [comment]
Apologies for the delay here. The local hashes all look good now!

```
Name           Hash
----           ----
jquery.js      54534731E9176376051FC2F721D23F8528863756DB0B50B80572C91FBE4F4688
jquery.min.js  412B8FF9C5AB32B9019FCD84BCD4A54C0E265A14528474F4EE45B27A20ABEAEB
jquery.min.map DB5EA0749F63CA5F4E2EB34D8F6F197D04BDDA1F633C72A0C77CB4BACFD484B1 
```

@HaoK will take a look to confirm we've got everything and then merge and handle the backporting.

**Comment by HaoK** [comment]
@mkArtakMSFT looks like there's a one other place that use jquery that isn't updated, MusicStore is using jquery 2.0.3, is this something we should update as well or is this only a test thing?

**Comment by wenz** [comment]
> @mkArtakMSFT looks like there's a one other place that use jquery that isn't updated, MusicStore is using jquery 2.0.3, is this something we should update as well or is this only a test thing?

AFAICS it's an ASP.NET MVC era sample app with additional dependencies that probably would not run with the current stack. I'd skip it, but just my two cents. Maybe @jongalloway knows its status?

**Comment by wenz** [comment]
Ugh, just noticed that the jQuery team released 3.5.0 in the meanwhile, and they mention that they fixed a security vulnerability in that release. I'll update my PR today or tomorrow. :-/

**Comment by HaoK** [comment]
Cool thanks @wenz yeah I wouldn't worry about music store then, we'll just get this in without it, since music store has some jquery-signalr in there too

**Comment by SIkebe** [comment]
jQery 3.5.0 has a breaking change and Bootstrap collapse doesn't work because of that.
https://github.com/jquery/jquery/issues/4665

So you might want to skip 3.5.0 and wait for 3.5.1 release.
https://github.com/jquery/jquery/milestone/23

**Comment by wenz** [comment]
Very good point @SIkebe, thanks! So we can either wait for jQuery 3.5.1 (not ETA, but several projects seem to be breaking with 3.5.0), or go ahead with jQuery 3.4.1 for now?

**Comment by mkArtakMSFT** [comment]
> 
> 
> @mkArtakMSFT looks like there's a one other place that use jquery that isn't updated, MusicStore is using jquery 2.0.3, is this something we should update as well or is this only a test thing?

We try to update dependencies everywhere, when we do so. So I'd update it too. Not a good practice to neglect part of the repo.

**Comment by wenz** [comment]
w/r/t MusicStore I'll give it a go this weekend (with jQuery 3.4.1 at first) - let's see how that goes.

**Comment by mkArtakMSFT** [comment]
Thanks @wenz.
I agree, 3.4.1 is a fine state to be in.

**Comment by wenz** [comment]
MvcMusicStore is updated to jQuery 3.4.1.

**Comment by HaoK** [comment]
@anurse do you know if its safe to use squash and merge on community prs?  I vaguely recall in the past it sometimes would lose giving proper credit in the commit history, just want to make sure its safe before I merge it that way

**Review by HaoK — APPROVED** [review]

**Comment by anurse** [comment]
It’s safe. GitHub will record the original author of the PR as the committer. There was a bug that broke this recently but it should be fixed now.

**Comment by wenz** [comment]
ah, don't worry about that! :)

**Comment by HaoK** [comment]
Thanks for doing this work @wenz !  I'll port this into 3.1/2.1

**Comment by wenz** [comment]
Thank you!

@mkArtakMSFT would you consider reopening #20354 - now that the insecure jQuery version is out of the way, we might now also want to consider updating other libraries (which are not insecure, but still not the latest version)?

**Pull request merged**